### PR TITLE
AEA-1730: Add dispense UI to EPSAT React client.

### PR DESCRIPTION
**Pull request by mfjarvis**

### CODE REVIEW

File: tool/site/client-react/src/components/buttonList.tsx [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import * as React from "react"
+import {CSSProperties, ReactNode} from "react"
+
+interface ButtonListProps {
+  children: ReactNode
+}
```
Comment (booshibooshi) [inline tool/site/client-react/src/components/buttonList.tsx:6]: is there anything about this component which restricts to buttons being passed in? looks more generic?

File: tool/site/client-react/src/components/buttonList.tsx [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import * as React from "react"
```
Comment (dan-jones-iw) [inline tool/site/client-react/src/components/buttonList.tsx:1]: combine react imports

File: tool/site/client-react/src/components/claim/claimPage.tsx [hunk 3]
```diff
@@ -70,10 +73,11 @@ const ClaimPage: React.FC<ClaimPageProps> = ({
       prescriptionDetails.medicationDispenses,
       claimFormValues
     )
-    const response = await axios.post(`${baseUrl}dispense/claim`, claim)
-    const responseStr = JSON.stringify(response.data)
+    const response = await axios.post<ClaimResult>(`${baseUrl}dispense/claim`, claim)
+    console.log(claim)
```
Comment (dan-jones-iw) [inline tool/site/client-react/src/components/claim/claimPage.tsx:77]: we still want these?

File: tool/site/client-react/src/components/buttonList.tsx (hunk 1 above)
Comment (booshibooshi) [reply tool/site/client-react/src/components/buttonList.tsx:6]: I guess the purpose for this right now is a buttonlist, can always rename if we find other uses for it, but happy it's named as its intended use for now

File: tool/site/client-react/src/components/claim/createDispenseClaim.ts [hunk 4]
```diff
@@ -108,67 +109,70 @@ function createClaimItem(
...
+}
+
 function createClaimItemDetail(
   sequence: number,
-  lineItemId: string,
   medicationRequest: fhir.MedicationRequest,
   medicationDispenses: Array<fhir.MedicationDispense>,
   productFormValues: ProductFormValues
 ): fhir.ClaimItemDetail {
+  const lineItemId = getMedicationRequestLineItemId(medicationRequest)
```
Comment (dan-jones-iw) [inline tool/site/client-react/src/components/claim/createDispenseClaim.ts:154]: we make the same call in `createClaimItem` and we also stopped passing it down into detail, any reason why?

File: tool/site/client-react/src/components/claim/claimPage.tsx (hunk 3 above)
Comment (mfjarvis) [reply tool/site/client-react/src/components/claim/claimPage.tsx:77]: Until we fix the fact that you don't get errors back from the backend

File: tool/site/client-react/src/components/claim/createDispenseClaim.ts (hunk 4 above)
Comment (mfjarvis) [reply tool/site/client-react/src/components/claim/createDispenseClaim.ts:154]: We're already passing it in (in the MedicationRequest)

File: tool/site/client-react/src/components/claim/createDispenseClaim.ts [hunk 5]
```diff
@@ -196,34 +200,32 @@ function createMedicationRequestReferenceExtension(lineItemId: string): ClaimMed
...
-}
-
-export function getMedicationDispenseLineItemId(medicationDispense: fhir.MedicationDispense): string {
-  return medicationDispense.authorizingPrescription[0].identifier.value
+function createEndorsementCodeableConcept(endorsement: EndorsementFormValues): fhir.CodeableConcept {
+  const endorsementCodeableConcept: fhir.CodeableConcept = {
+    coding: VALUE_SET_DISPENSER_ENDORSEMENT.filter(coding => coding.code === endorsement.code)
+  }
+  if (endorsement.supportingInfo) {
+    endorsementCodeableConcept.text = endorsement.supportingInfo
```
Comment (dan-jones-iw) [inline tool/site/client-react/src/components/claim/createDispenseClaim.ts:228]: ew

File: tool/site/client-react/src/components/dispense/createDispenseNotification.ts [hunk 6]
```diff
@@ -0,0 +1,265 @@
...
+    return createMedicationDispense(medicationRequest, formValuesForLineItem, dispenseFormValues.prescription)
+  })
+
+  //TODO - Fix resources without IDs in examples? Or just assign a fresh ID here
+  if (!patient.id) {
+    patient.id = medicationRequests[0].subject.reference.substring("urn:uuid:".length)
+  }
+
+  //TODO - work out why we're getting validation errors
+  patient.identifier[0] = {
```
Comment (dan-jones-iw) [inline tool/site/client-react/src/components/dispense/createDispenseNotification.ts:56]: what is happening here?

Comment (mfjarvis) [reply tool/site/client-react/src/components/dispense/createDispenseNotification.ts:56]: Something about nhs number verification status

Comment (mfjarvis) [reply tool/site/client-react/src/components/dispense/createDispenseNotification.ts:56]: Haven't looked into it yet

File: tool/site/client-react/src/components/prescription-summary/newLineFormatter.tsx [hunk 7]
```diff
@@ -1,5 +1,5 @@
-import {Fragment} from "react"
 import * as React from "react"
+import {Fragment} from "react"
```
Comment (dan-jones-iw) [inline tool/site/client-react/src/components/prescription-summary/newLineFormatter.tsx:2]: :sob:

Comment (mfjarvis) [reply tool/site/client-react/src/components/prescription-summary/newLineFormatter.tsx:2]: Go away

File: tool/site/client-react/src/components/claim/createDispenseClaim.ts (hunk 5 above)
Comment (mfjarvis) [reply tool/site/client-react/src/components/claim/createDispenseClaim.ts:227]: Agree

**Review by dan-jones-iw — APPROVED** [review]

**Pull request merged**